Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -773,6 +765,14 @@ void ContainerNode::removeBetween(Node* previousChild, Node* nextChild, Node& ol
     oldChild.setParentNode(nullptr);
 
     oldChild.setTreeScopeRecursively(document());
+
+    auto* element = dynamicDowncast<Element>(oldChild);
```

Comment:
Can we add this to Element::removedFromAncestor instead?
We can check `!parentNode()` to detect the top-most node in a removed subtree.

Reply:
Yeah but generally, removedFromAncestor is where we update these kinds of caches so it's more appropriate for it to be there. Besides that, removeBetween is a core DOM mutation helper so logic specific to IntersectionObserver doesn't really belong there.

My commentary about `!parentNode()` is that `removedFromAncestor` gets called on a node when it was removed from its immediate parent, or its ancestor is removed from its parent. So in order to run the code only when its immediate parent is changed, we need to check `!parentNode()` so your code in `Element::removedFromAncestor` will look like this:
```
if (!parentNode() && (lastRememberedLogicalWidth() || lastRememberedLogicalHeight())) {
    document().observeForContainIntrinsicSize(*this);
    document().resetObservationSizeForContainIntrinsicSize(*this);
}
```

Reply:
@rniwa I see, thanks for the explanation!